mobilenet-burn/src/model/mobilenet.rs

```diff
@@ -0,0 +1,244 @@
+            ),
+        ];
+
+        MobileNetV2 {
+            features,
+            classifier,
+            avg_pool: AdaptiveAvgPool2dConfig::new([1, 1]).init(),
+        }
+    }
+    // fn init_with<B: Backend>(&self, record: MobileNetV2Record<B>) -> MobileNetV2<B> {
+    //     let num_features = record.features.len();
+    //     let features = (0..num_features)
+    //         .zip(record.features)
+    //         .map(|(b, r)| match r {
+    //             FeatureLayersType::Block(block) => FeatureLayersType::Block(block.init_with(&r)),
+    //             FeatureLayersType::Conv(conv) => FeatureLayersType::Conv(conv.init_with(&r)),
+    //         })
+    //         .collect();
+    //     let num_classifier = record.classifier.len();
+    //     let classifier = (0..num_classifier)
+    //         .zip(record.classifier)
+    //         .map(|b, r| match r {
+    //             ClassifierLayersType::Dropout(dropout) => {
+    //                 ClassifierLayersType::Dropout(dropout.init_with(&r))
+    //             }
+    //             ClassifierLayersType::Linear(linear) => {
+    //                 ClassifierLayersType::Linear(linear.init_with(&r))
+    //             }
+    //         })
+    //         .collect();
+    //
+    //     MobileNetV2 {
+    //         features,
+    //         classifier,
+    //         avg_pool: AdaptiveAvgPool2dConfig::new([1, 1]).init(),
+    //     }
+    // }
+}
+impl<B: Backend> MobileNetV2<B> {
+    pub fn forward(&self, input: Tensor<B, 4>) -> Tensor<B, 4> {
+        let mut x = input;
+        for layer in &self.features {
+            match layer {
+                ConvBlock::InvertedResidual(block) => {
+                    x = block.forward(&x);
+                }
+                ConvBlock::Conv(conv) => {
+                    x = conv.forward(x);
+                }
+            }
+        }
+
+        x = self.avg_pool.forward(x);
+        x = x.flatten(1, 1);
+        for layer in &self.classifier {
+            match layer {
+                ClassifierLayersType::Dropout(dropout) => {
+                    x = dropout.forward(x);
+                }
+                ClassifierLayersType::Linear(linear) => {
+                    x = linear.forward(x);
+                }
+            }
+        }
+        x
+    }
+}
+
+// #[cfg(feature = "pretrained")]
+// impl<B: Backend> MobileNetV2<B> {
+//     /// Load specified pre-trained PyTorch weights as a record.
+//     fn load_weights_record(
+//         weights: &weights::Weights,
+//         device: &Device<B>,
+//     ) -> Result<MobileNetV2Record<B>, RecorderError> {
+//         // Download torch weights
+//         let torch_weights = weights.download().map_err(|err| {
+//             RecorderError::Unknown(format!("Could not download weights.\nError: {err}"))
+//         })?;
+//
+//         // Load weights from torch state_dict
+//         let load_args = LoadArgs::new(torch_weights)
```

Comment:
this was just copied from resnet XD

Reply:
I can take over after the other comments are addressed to add the weights loading. It'll probably follow a very similar pattern to the resnet, the most important part will be remapping the keys properly 🙂